### PR TITLE
Adding support for custom hosted CDN's 

### PR DESCRIPTION
**Pull request by AlisterGreg**
CascLib currently only supports connection to existing blizzard CDN's. Within these commits I've added support for a custom connection URL to be supplied that will enable the ability to connect to privately hosted CDN's.

Summary of changes:
-Pushed the hardcoded ribbit/Http URL up so that It can now be passed in via arguments.
-Added support to parse the port from custom URL's as the port may differ from 80 for privately hosted CDN's.

### CODE REVIEW

**Comment by AlisterGreg** [comment]
That's no problem, I have a couple to test, noticed that the region is hardcoded so I'll push that up as well, once I have these working ill send and email across!

File: src/CascLib.h [hunk 1]
```diff
@@ -316,6 +316,9 @@ typedef struct _CASC_OPEN_STORAGE_ARGS
 {
     size_t Size;                                // Length of this structure. Initialize to sizeof(CASC_OPEN_STORAGE_ARGS)
 
+    LPCTSTR szCdnHostUrl;                       // Online: Specify the cdn host url
+    LPCTSTR szCdnHostRegion;                    // Online: Specify the cdn host region
```
Comment (Shauren) [inline src/CascLib.h:320]: For `ExtractVersionedArgument` to work properly, all new members must always be added at the end of the structure

Comment (ladislav-zezula) [reply src/CascLib.h:320]: @Shauren Yep. That's why in my [incoming pull request](https://github.com/ladislav-zezula/CascLib/blob/53d417a21ce7529b65d4fae9dab3b3d360a7162d/src/CascLib.h#L344), it's been placed at the end of the structure.

**Comment by ladislav-zezula** [comment]
Closing the pull request. The changes were done in [alternate pull request](https://github.com/ladislav-zezula/CascLib/pull/211). Please, put all eventual feedback there.

**Pull request closed without merging**